### PR TITLE
MBS-11916 fix some dependencies was not downloaded through proxy

### PR DESCRIPTION
**Pull request by dsvoronin**
Checked by running with `--refresh-dependencies` and looking to scan / network page

Problem was combined:
not only i flushed all the caches
jcenter and gradle plugins portal based on it's repos degraded: https://status.gradle.com/

### CODE REVIEW

File: subprojects/build.gradle.kts [hunk 1]
```diff
@@ -2,7 +2,7 @@ plugins {
     id("convention.lifecycle")
     // accessing version catalog here is blocked by IDE false-postive error
     // https://youtrack.jetbrains.com/issue/KTIJ-19369
-    id("com.autonomousapps.dependency-analysis") version "0.74.0"
+    id("com.autonomousapps.dependency-analysis") version "0.77.0"
```
Comment (eugene-krivobokov) [inline subprojects/build.gradle.kts:5]: Is it a necessary update for the fix?

**Review by eugene-krivobokov — APPROVED** [review]

File: subprojects/settings.gradle.kts [hunk 2]
```diff
@@ -1,5 +1,76 @@
 enableFeaturePreview("VERSION_CATALOGS")
-enableFeaturePreview("TYPESAFE_PROJECT_ACCESSORS")
+
+pluginManagement {
+    includeBuild("../build-logic-settings")
+
+    val artifactoryUrl: String? by settings
+
+    repositories {
+        maven {
```
Comment (sergioboi) [inline subprojects/settings.gradle.kts:9]: Could we make an extension function `mavenWithProxy` that will have 3 parameters: title, URL, proxy URL
To reduce duplications

**Review by sergioboi — APPROVED** [review]

File: subprojects/build.gradle.kts (hunk 1 above)
Comment (dsvoronin) [reply subprojects/build.gradle.kts:5]: Plugin was failing with weird stacktrace until i updated it

File: subprojects/settings.gradle.kts (hunk 2 above)
Comment (dsvoronin) [reply subprojects/settings.gradle.kts:9]: looks like not in pluginManagement

**Comment by dsvoronin** [comment]
https://github.com/avito-tech/avito-android/pull/1224

**Pull request closed without merging**